elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/CompletionDsl.scala

```diff
@@ -33,7 +33,35 @@ trait CompletionDsl extends DslCommons with QueryDsl {
     }
   }
 
-  case class Completion(field: String, size: Int, context: Map[String, String]) extends Query {
+  case class Completion(field: String, size: Int, context: Map[String, String])
+    extends Query with CompletionWithContext {
+
+    val _field = "field"
+    val _context = "context"
+    val _size = "size"
+
+    def withAdditionalContext(newContext: (String, String)*) = {
+      this.copy(context = context ++ newContext)
+    }
+
+    override def toJson: Map[String, Any] = Map(
+      _field -> field,
+      _context -> context,
+      _size -> size
+    )
+  }
+
+  trait CompletionWithContext {
+    def toJson: Map[String, Any]
```

Comment:
It should be possible to share the code between these 2 classes. 

```
trait CompletionWithContext {
   val _field = "field"
  def context: Iterable[(String, String)]
  def toJson: Map[String, Any] = ...
}
```

That reduces the chance of introducing bugs by changing 1 and not the other.